### PR TITLE
Fixed #33057 -- Fixed recreation of foreign key constraints in m2m tables when altering type of referenced primary key on Oracle.

### PR DESCRIPTION
**Pull request by felixxm**
See [comment](https://code.djangoproject.com/ticket/33057#comment:2).

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/backends/oracle/schema.py [hunk 1]
```diff
@@ -124,6 +126,19 @@ def _alter_field_type_workaround(self, model, old_field, new_field):
         self.remove_field(model, old_field)
         # Rename and possibly make the new field NOT NULL
         super().alter_field(model, new_temp_field, new_field)
+        # Recreate foreign key (if necessary) because the old field is not
+        # passed to the alter_field() and data types of new_temp_field and
+        # new_field always match.
+        new_type = new_field.db_type(self.connection)
+        if (
+            self.connection.features.supports_foreign_keys and (
```
Comment (timgraham) [inline django/db/backends/oracle/schema.py:134]: Is there a purpose in consulting this flag that's always True on Oracle?

Comment (felixxm) [reply django/db/backends/oracle/schema.py]: Eagle eye :eagle: Thanks, removed.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Review by carltongibson — APPROVED** [review]
OK, thanks. 👍

**Comment by felixxm** [comment]
Thanks both :1st_place_medal:

**Pull request merged**